### PR TITLE
added solution for bus tickets hw

### PR DESCRIPTION
**Pull request by AnyaRatusnyak**

### CODE REVIEW

File: src/TicketService.java [hunk 1]
```diff
@@ -22,7 +24,7 @@ public static void main(String[] args) {
             Ticket ticketById = findById(searchId, tickets);
             System.out.println("Found ticket: " + ticketById);
         } catch (NoSuchElementException e) {
-            System.out.println("Ticket with id " + searchId + " not found.");
+            System.out.println("model.Ticket with id " + searchId + " not found.");
```
Comment (Alexmansar) [inline src/TicketService.java]: **model.** why we need it?

File: src/model/Admin.java [hunk 2]
```diff
@@ -1,11 +1,13 @@
...
+
 public class Admin extends User{
     public Admin(int id) {
         super(id);
     }
 
     @Override
     public void printRole() {
-        System.out.println("I am Admin ");
+        System.out.println("I am model.Admin ");
```
Comment (Alexmansar) [inline src/model/Admin.java]: the same

File: src/model/BusTicket.java [hunk 3]
```diff
@@ -0,0 +1,51 @@
+package model;
+
+public class BusTicket extends Ticket{
+private TicketClass ticketClass;
+private TicketType ticketType;
+private String startDate;
+private int busTicketPrice;
```
Comment (Alexmansar) [inline src/model/BusTicket.java]: please add spaces))

File: src/model/BusTicket.java [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+                "ticketClass=" + ticketClass +
+                ", ticketType=" + ticketType +
+                ", startDate='" + startDate + '\'' +
+                ", busTicketPrice=" + busTicketPrice +
+                ", id=" + id +
+                '}';
+    }
+
+    public enum TicketClass {
+CLA, STD
```
Comment (Alexmansar) [inline src/model/BusTicket.java]: here as well
ctrl+alt+l 
in IDE

File: src/util/TicketValidator.java [hunk 5]
```diff
@@ -0,0 +1,85 @@
...
+            String error = entry.getKey();
+            int count = entry.getValue();
+            if (count > maxCount) {
+                mostFrequentError = error;
+                maxCount = count;
+            }
+        }
+
+        return mostFrequentError;
+    }
```
Comment (Alexmansar) [inline src/util/TicketValidator.java]: move it before our public methods)

**Review by Alexmansar — CHANGES_REQUESTED** [review]
@AnyaRatusnyak Hi, please fix comments)) and remove in all cases **model.**

File: src/TicketService.java (hunk 1 above)
Comment (AnyaRatusnyak) [reply src/TicketService.java]: Hi! Sorry! It was automatic after I moved my classes to the new model folder. Thank you a lot ! I didn't notice.

File: src/util/TicketValidator.java [hunk 6]
```diff
@@ -0,0 +1,87 @@
...
+
+    public boolean validateBusTicket(BusTicket busTicket) {
```
Comment (Alexmansar) [inline src/util/TicketValidator.java]: can you devide this big method to smallers?

**Review by Alexmansar — CHANGES_REQUESTED** [review]
@AnyaRatusnyak One more request ))

File: src/util/TicketValidator.java (hunk 6 above)
Comment (AnyaRatusnyak) [reply src/util/TicketValidator.java]: Sure! Thanks)

**Review by Alexmansar — APPROVED** [review]
@AnyaRatusnyak Could be merged)

**Pull request merged**